torch/utils/data/dataset.py

```diff
@@ -225,6 +226,11 @@ def __getattr__(self, attribute_name):
         else:
             raise AttributeError
 
+    def __getstate__(self):
+        if IterableDataset.getstate_hook is not None:
+            return IterableDataset.getstate_hook(self)
+        return self.__dict__
+
```

Comment:
I really want these APIs are moved out of `IterableDataset`. I mean making a new `IterDataPipe` derived from `IterableDataset`.